beacon-chain/blockchain/chain_info.go

```diff
@@ -332,7 +334,17 @@ func (s *Service) IsOptimisticForRoot(ctx context.Context, root [32]byte) (bool,
 		return false, err
 	}
 	if ss == nil {
-		return false, errInvalidNilSummary
+		// if the requested root is the headroot we should treat the
+		// node as optimistic. This can happen if we pruned INVALID
+		// nodes and no viable head is available.
+		headRoot, err := s.HeadRoot(ctx)
```

Comment:
how does this work, we have a head for which there is no state summary/block in the db ?

Reply:
Yes, and even worse, it can be different than forkchoice's